File: bmmo/blocks_table.py
class BlocksTable():
    def __init__(self, type: str, sound: str, table: list[list[int]] = None):
        if table is None:
            table = []
        self.type = type
        self.sound = sound
        self.table = list(table)

    def __getitem__(self, item: int):
        return self.table[item]

    def __setitem__(self, key: int, value: int):
        self.table[key] = value

    def __repr__(self):
        return f'Type: {self.type}\n Sound: {self.sound}\n Table: {str(self.table)}\n'

File: src/bmmo/blocks_table.ts
export class BlocksTable {
  type: string;
  sound: string;
  table: number[][];

  constructor(type: string, sound: string, table: number[][] | null = null) {
    this.type = type;
    this.sound = sound;
    this.table = table ? [...table] : [];
  }

  get(item: number): number[] {
    return this.table[item];
  }

  set(key: number, value: number[]): void {
    this.table[key] = value;
  }

  toString(): string {
    return `Type: ${this.type}\n Sound: ${this.sound}\n Table: ${JSON.stringify(
      this.table
    )}\n`;
  }
}
